command/agent/retry_join_test.go

```diff
@@ -165,15 +158,46 @@ func TestRetryJoin_MixedProvider(t *testing.T) {
 
 	mockDiscover := &MockDiscover{}
 	joiner := retryJoiner{
-		discover: mockDiscover,
-		join:     mockJoin,
-		logger:   log.New(ioutil.Discard, "", 0),
-		errCh:    make(chan struct{}),
+		discover:      mockDiscover,
+		serverJoin:    mockJoin,
+		serverEnabled: true,
+		logger:        log.New(ioutil.Discard, "", 0),
+		errCh:         make(chan struct{}),
 	}
 
-	joiner.RetryJoin(newConfig)
+	joiner.RetryJoin(serverJoin)
 
 	require.Equal(2, len(output))
 	require.Equal("provider=aws, tag_value=foo", mockDiscover.ReceivedAddrs)
 	require.Equal(stubAddress, output[0])
 }
+
+func TestRetryJoin_Client(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	serverJoin := &ServerJoin{
+		RetryMaxAttempts: 1,
+		RetryJoin:        []string{"127.0.0.1"},
+	}
+
+	var output []string
+
+	mockJoin := func(s []string) (int, error) {
+		output = s
+		return 0, nil
+	}
+
+	joiner := retryJoiner{
+		discover:      &MockDiscover{},
+		clientJoin:    mockJoin,
+		clientEnabled: true,
+		logger:        log.New(ioutil.Discard, "", 0),
+		errCh:         make(chan struct{}),
+	}
+
+	joiner.RetryJoin(serverJoin)
+
+	require.Equal(1, len(output))
+	require.Equal(stubAddress, output[0])
+}
```

Comment:
Can we do a client and server test

Reply:
I changed an existing test to be server-specific, but I'll mark it in the test name as such.